FWCore/Utilities/interface/RandomNumberGenerator.h

```diff
@@ -157,6 +158,8 @@ namespace edm {
     /// Use this engine in the global begin luminosity block method
     virtual CLHEP::HepRandomEngine& getEngine(LuminosityBlockIndex const&) = 0;
 
+    virtual std::unique_ptr<CLHEP::HepRandomEngine> cloneEngine(LuminosityBlockIndex const&) = 0;
```

Comment:
Could you add a short comment explaining the intended use case (including something around "not intended for general use")?

Reply:
Done. Good  idea.